cli/src/main/kotlin/utils/ConfigDirHelper.kt

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (C) 2025 The ORT Server Authors (See <https://github.com/eclipse-apoapsis/ort-server/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.eclipse.apoapsis.ortserver.cli.utils
+
+import java.io.File
+
+import org.eclipse.apoapsis.ortserver.cli.COMMAND_NAME
+
+import org.ossreviewtoolkit.utils.common.Os
+import org.ossreviewtoolkit.utils.common.safeMkdirs
+
+/**
+ * The directory where the configuration files are stored, respecting the XDG Base Directory specification [1].
+ *
+ * [1]: https://specifications.freedesktop.org/basedir-spec/latest/
+ */
+internal val configDir: File
+    get() {
+        val dir = when {
+            Os.isLinux -> File(
+                System.getenv("XDG_CONFIG_HOME") ?: "${Os.fixupUserHomeProperty()}/.config/$COMMAND_NAME"
+            )
+
+            Os.isMac -> File(System.getenv("XDG_CONFIG_HOME") ?: "${Os.fixupUserHomeProperty()}/.config/$COMMAND_NAME")
+
+            Os.isWindows -> File(
+                System.getenv("XDG_CONFIG_HOME")
+                    ?: "${
+                        System.getenv("LOCALAPPDATA")
+                            ?: "${Os.fixupUserHomeProperty()}/.config"
+                    }/$COMMAND_NAME"
+            )
+
+            else -> File("${Os.fixupUserHomeProperty()}/.config/$COMMAND_NAME")
+        }
+
+        if (!dir.exists()) dir.safeMkdirs()
+
+        return dir
+    }
+
+/**
+ * The file where the authentications are stored.
+ */
+internal val authFile: File
```

Comment:
Would this property better belong to `AuthenticationUtils`?

In general, I have some difficulties to understand the various components spread over multiple projects that are involved in storing authentication data. Exposing the `authFile` here reminds me of the concept of global variables in pre-OO times.

As a suggestion, could there be an `object AuthenticationStorage` in the `cli` module that encapsulates all the internal data types and details? It could expose an API on a more higher level, to query and set authentication data without clients have to deal with file access.

Also, is it really necessary that the model for storing authentication data is part of the `api` module?